src/openrct2/ride/Vehicle.cpp

```diff
@@ -8028,6 +8038,109 @@ bool Vehicle::UpdateTrackMotionForwardsGetNewTrack(uint16_t trackType, Ride* cur
     return true;
 }
 
+int32_t Vehicle::CalculateRiderBraking() const
+{
+    int32_t mode = rider_speed_preference & 0x3F;
+
+    int32_t straightTargetSpeed[] = {
```

Comment:
Make this into a struct for the settings. Move the struct and settings to a dedicated file. I would also consider moving this function to that file. Vehicle.cpp is already far too busy as is and since this is all just for one item it will make for an easy to define interface.